src/zenml/integrations/mlflow/__init__.py

```diff
@@ -35,9 +35,9 @@ class MlflowIntegration(Integration):
     # does not pin it. They fixed this in a later version, so we can probably
     # remove this once we update the mlflow version.
     REQUIREMENTS = [
-        "mlflow>=1.24.0,<=2.2.2",
-        "mlserver>=0.5.3",
-        "mlserver-mlflow>=0.5.3",
+        "mlflow>=2.1.1,<=2.5.0",
```

Comment:
Why do we go for `2.1.1` here? What makes `2.1.1` better than `2.0.1`?
If limiting the lower boundary above the earliest `2x` version, I'd go for `2.2.1` as it was delivering important fixes.

Reply:
The reason for `2.1.1` is a bug fixed in that version related to the model registry so the previous version would still have that bug.

Reply:
For my understanding - which [one of those](https://github.com/mlflow/mlflow/releases/tag/v2.1.1) is critical for our case?

Reply:
I think it's more of a feature within mlflow rather than a bug but for the zenml model registry, it was detected as a bug. 
At the development of the model registry stack component, it was designed so that it uses the same backend store as the experiment tracker, in a local case our mlflow experiment tracker uses the file store [this one](https://github.com/mlflow/mlflow/blob/branch-2.1/mlflow/store/tracking/file_store.py) However before version `2.1.x` of MLflow there was no implementation of a backend file store for the registry there were only [rest_store](https://github.com/mlflow/mlflow/blob/branch-2.1/mlflow/store/model_registry/rest_store.py) and [sqlalchemy_store](https://github.com/mlflow/mlflow/blob/branch-2.1/mlflow/store/model_registry/rest_store.py). The absence of a file_store was causing an error because within zenml we were trying to configure both experiment tracker and model registry to use the same store and save all artifact in the same path, however starting from the version 2.1 the file_store was introduced to the model registry which you can find [here](https://github.com/mlflow/mlflow/blob/branch-2.1/mlflow/store/model_registry/file_store.py) which was also mentioned in the release notes from [here](https://github.com/mlflow/mlflow/releases/tag/v2.1.0). With how coupled experiment tracking and registry components are (it's not set to change, you can see an open PR [here](https://github.com/zenml-io/zenml/pull/1549)) it was a challenging task to make the registry support an older version that doesn't have a file_store so for that we just limited the version to 2.1 only for when the model registry is used which you can find from [here](https://github.com/zenml-io/zenml/blob/main/src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py#L110)

Reply:
Technically we can still have the validation we do for verifying if a local model registry is used and the version installed is earlier than `2.1.x` and have a range of supported mlflow versions starting from`2.0` @avishniakov do you think that would be better? or we just move the `2.2.1` which has even more important fixes as you mentioned?

Reply:
I see, thanks for the clarifications!
My suggestion would be to go with `2.1.1` as a minimum then, but raise a warning in case `<2.2.1` is used, so users are better aware of security concerns it might bring. Sounds reasonable?